libs/os/src/os_time.c

```diff
@@ -141,3 +141,19 @@ os_gettimeofday(struct os_timeval *tv, struct os_timezone *tz)
 
     return (0);
 }
+
+int64_t
+os_get_uptime_usec(void) {
+  struct os_timeval tv;
+  os_time_t delta;
+  os_sr_t sr;
+
+  OS_ENTER_CRITICAL(sr);
+
+  delta = os_time_get() - basetod.ostime;
+  os_deltatime(delta, &basetod.uptime, &tv);
```

Comment:
This is a long period to disable a critical section.  can you just grab basetod.os_time, then do the calculation?